jsonrpc/specification.md

```diff
@@ -0,0 +1,49 @@
+# Portal JSON-RPC specification
+
+The `*` is meant to be a placeholder to substitute different Portal sub-networks `history`, `state`, `beacon`, etc
+
+## portal_*RoutingTableInfo
+
+## portal_*AddEnr
+- Adds the Enr to the respective sub-networks routing table
+
+## portal_*GetEnr
+
+## portal_*DeleteEnr
+
+## portal_*LookupEnr
+
+## portal_*Ping
+
+## portal_*FindNodes
+
+## portal_*FindContent
+- must validate content
+
+## portal_*Offer
+- must not validate content
+- shouldn't store data on offering node
+
+## portal_*RecursiveFindNodes
+
+## portal_*GetContent
+- must validate content
+- must check local storage first before doing recursive find content
+- if recursive find content is done must do poke mechanism
+- if recursive find content is done and content meets storage criteria store content
+  
+## portal_*TraceGetContent
+- must validate content
+- must check local storage first before doing recursive find content
+- if recursive find content is done must do poke mechanism
+- if recursive find content is done and content meets storage criteria store content
+
+## portal_*Store
+- must not validate content
+  
+## portal_*LocalContent
+- content stored locally is assumed already validated
+
+## portal_*Gossip
+- must not validate content
+- shouldn't store data on gossiping node
```

Comment:
At Fluffy we were thinking of having validation + storage on this call but not on the `portal_*Offer` considering that one seems lower level. But we are not sure about it ourselves, so works either way for us.

Reply:
> At Fluffy we were thinking of having validation

If someone gossips invalid data the node gossiped to should temporarily ban the gossiper.

I don't think you can validate everything without making network requests

Reply:
In Fluffy we do local validations without doing network requests (currently just the state network). Sure, the node receiving the offer will validate and reject the data if it is bad but if you can prevent the bad data before sending then why not.